src/components/Trade/TradeConfirm/TradeConfirm.tsx

```diff
@@ -73,9 +75,19 @@ export const TradeConfirm = ({ history }: RouterProps) => {
     } catch (err) {
       console.error(`TradeConfirm:onSubmit - ${err}`)
       // TODO: (ryankk) this needs to be revisited post bounty to handle actual errors coming back from unchained.
+      let errorMessage
+      switch ((err as SwapError)?.message) {
+        case 'ZrxExecuteQuote - signAndBroadcastTransaction error: Error: Error signing & broadcasting tx': {
+          errorMessage = TRADE_ERRORS.TRANSACTION_REJECTED
+          break
+        }
+        default: {
+          errorMessage = TRADE_ERRORS.INSUFFICIENT_FUNDS
```

Comment:
We want to target specifically `signAndBroadcastTransaction` errors. There might be other errors caught here, which are not a result of user rejection.